src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -218,13 +213,96 @@ public Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document,
+
+            if (hasAnyProjectFixer && projectFixersMap.TryGetValue(diagnosticId, out var projectFixers))
+            {
+                Debug.Assert(!isInteractive);
+                allFixers.AddRange(projectFixers);
+            }
+
+            var primaryDiagnostic = await diagnostics[0].ToDiagnosticAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            var extensionManager = document.Project.Solution.Workspace.Services.GetService<IExtensionManager>();
+
+            var result = ArrayBuilder<CodeFixCollection>.GetInstance();
+
+            foreach (var fixer in allFixers.Distinct())
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var fixes = await extensionManager.PerformFunctionAsync(fixer,
+                    () => GetCodeFixesAsync(document, primaryDiagnostic.Location.SourceSpan, fixer, ImmutableArray.Create(primaryDiagnostic), cancellationToken),
+                    defaultValue: ImmutableArray<CodeFix>.Empty).ConfigureAwait(false);
+
+                if (fixes.IsDefaultOrEmpty)
+                {
+                    continue;
+                }
+
+                // If the fix provider supports fix all occurrences, then get the corresponding FixAllProviderInfo and fix all context.
+                var fixAllProviderInfo = extensionManager.PerformFunction(fixer, () => ImmutableInterlocked.GetOrAdd(ref _fixAllProviderMap, fixer, FixAllProviderInfo.Create), defaultValue: null);
+                if (fixAllProviderInfo == null)
+                {
+                    // for now, we don't care about fixer that doesn't have fix all implementation
+                    return null;
+                }
+
+                var diagnosticProvider = new FixAllPredefinedDiagnosticProvider(await diagnostics.ToDiagnosticsAsync(document.Project, cancellationToken).ConfigureAwait(false));
+                var fixAllState = new FixAllState(
+                    fixAllProvider: fixAllProviderInfo.FixAllProvider,
+                    document: document,
+                    codeFixProvider: fixer,
+                    scope: FixAllScope.Document,
+                    codeActionEquivalenceKey: fixes.First().Action.EquivalenceKey,
+                    diagnosticIds: SpecializedCollections.SingletonEnumerable(diagnosticId),
+                    fixAllDiagnosticProvider: diagnosticProvider);
+
+                var codeFix = new CodeFixCollection(
+                    fixer, primaryDiagnostic.Location.SourceSpan, fixes, fixAllState,
+                    fixAllProviderInfo.SupportedScopes, primaryDiagnostic);
+
+                result.Add(codeFix);
+                break;
+            }
 
             // TODO: Just get the first fix for now until we have a way to config user's preferred fix
             // https://github.com/dotnet/roslyn/issues/27066
-            return fixesCollectionArray.FirstOrDefault();
+            return result.ToImmutableAndFree().FirstOrDefault();
```

Comment:
if we are going to just take first. why not just return when we have first codefix? rather than add all those in result and then throw away?